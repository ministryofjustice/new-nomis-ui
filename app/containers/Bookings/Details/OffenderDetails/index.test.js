import React from 'react'
import renderer from 'react-test-renderer'
import { IntlProvider } from 'react-intl'
import Immutable from 'immutable'
import { OffenderDetails } from './index'

describe('<OffenderDetails />', () => {
  const offenderDetails = Immutable.Map(
    Immutable.fromJS({
      physicalAttributes: {
        gender: 'Male',
        raceCode: 'W1',
        ethnicity: 'White: Eng./Welsh/Scot./N.Irish/British',
        heightFeet: 5,
        heightInches: 9,
        heightMetres: 1.75,
        heightCentimetres: 175,
        weightPounds: 168,
        weightKilograms: 76,
        sexCode: 'M',
      },
      physicalCharacteristics: [
        {
          type: 'HAIR',
          characteristic: 'Hair Colour',
          detail: 'QBNQB',
        },
        {
          type: 'R_EYE_C',
          characteristic: 'Right Eye Colour',
          detail: 'AIAI',
        },
        {
          type: 'BUILD',
          characteristic: 'Build',
          detail: 'YEkYEk',
        },
      ],
      physicalMarks: [
        {
          type: 'Tattoo',
          side: 'Left',
          bodyPart: 'Torso',
          comment: 'INIgBKyJIPuINIgBKyJIP',
          orentiation: 'Centre',
        },
      ],
      dateOfBirth: '1980-01-02',
      age: '40',
      language: 'English',
      profileInformation: [{ type: 'RELF', resultValue: 'Atheist' }, { type: 'NAT', resultValue: 'British' }],
      primaryAddress: {
        flat: '7',
        premise: 'premises',
        street: 'street name',
        town: 'Barnsley',
        postalCode: 'LS1 XXX',
        county: 'South Yorkshire',
        country: 'England',
        comment: 'comment text goes here',
        primary: true,
        noFixedAddress: false,
        type: 'PRESENT',
      },
      aliases: [],
    })
  )

  it('should match the default snapshot', () => {
    const tree = renderer
      .create(
        <IntlProvider locale="en">
          <OffenderDetails offenderDetails={offenderDetails} showPhoto={jest.fn} />
        </IntlProvider>
      )
      .toJSON()

    expect(tree).toMatchSnapshot()
  })
})
