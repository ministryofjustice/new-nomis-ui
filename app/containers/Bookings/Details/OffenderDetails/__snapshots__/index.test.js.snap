// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<OffenderDetails /> should match the default snapshot 1`] = `
<div
  className="offender-details"
>
  <div
    className="row"
  >
    <div
      className="col-md-6"
    >
      <div
        className="row"
      >
        <h3
          className="heading-medium top-heading"
        >
          Personal details
        </h3>
      </div>
      <div
        className="sc-bwzfXH ijiype"
      >
        <div
          className="sc-htpNat idItfP"
        >
          Date of birth
        </div>
        <div
          className="sc-htpNat idItfP"
        >
          <strong>
            <span>
              01/02/1980
            </span>
          </strong>
        </div>
      </div>
      <div
        className="sc-bwzfXH ijiype"
      >
        <div
          className="sc-htpNat idItfP"
        >
          Age
        </div>
        <div
          className="sc-htpNat idItfP"
        >
          <strong>
            40
          </strong>
        </div>
      </div>
      <div
        className="sc-bwzfXH ijiype"
      >
        <div
          className="sc-htpNat idItfP"
        >
          Gender
        </div>
        <div
          className="sc-htpNat idItfP"
        >
          <strong>
            Male
          </strong>
        </div>
      </div>
      <div
        className="sc-bwzfXH ijiype"
      >
        <div
          className="sc-htpNat idItfP"
        >
          Ethnicity
        </div>
        <div
          className="sc-htpNat idItfP"
        >
          <strong>
            White: Eng./Welsh/Scot./N.Irish/British (W1)
          </strong>
        </div>
      </div>
      <div
        className="sc-bwzfXH ijiype"
      >
        <div
          className="sc-htpNat idItfP"
        >
          Religion or belief
        </div>
        <div
          className="sc-htpNat idItfP"
        >
          <strong>
            Atheist
          </strong>
        </div>
      </div>
      <div
        className="sc-bwzfXH ijiype"
      >
        <div
          className="sc-htpNat idItfP"
        >
          Nationality
        </div>
        <div
          className="sc-htpNat idItfP"
        >
          <strong>
            British
          </strong>
        </div>
      </div>
      <div
        className="sc-bwzfXH ijiype"
      >
        <div
          className="sc-htpNat idItfP"
        >
          Spoken language
        </div>
        <div
          className="sc-htpNat idItfP"
        >
          <strong>
            English
          </strong>
        </div>
      </div>
      <div
        className="row"
      >
        <div
          className="col-md-12"
        >
          <h3
            className="heading-medium"
          >
            Primary Address
          </h3>
          <div
            className="sc-bwzfXH ijiype"
          >
            <div
              className="sc-htpNat idItfP"
            >
              Flat
            </div>
            <div
              className="sc-htpNat idItfP"
            >
              <strong>
                7
              </strong>
            </div>
          </div>
          <div
            className="sc-bwzfXH ijiype"
          >
            <div
              className="sc-htpNat idItfP"
            >
              Street Address
            </div>
            <div
              className="sc-htpNat idItfP"
            >
              <strong>
                premises, street name
              </strong>
            </div>
          </div>
          <div
            className="sc-bwzfXH ijiype"
          >
            <div
              className="sc-htpNat idItfP"
            >
              Town
            </div>
            <div
              className="sc-htpNat idItfP"
            >
              <strong>
                Barnsley
              </strong>
            </div>
          </div>
          <div
            className="sc-bwzfXH ijiype"
          >
            <div
              className="sc-htpNat idItfP"
            >
              County
            </div>
            <div
              className="sc-htpNat idItfP"
            >
              <strong>
                South Yorkshire
              </strong>
            </div>
          </div>
          <div
            className="sc-bwzfXH ijiype"
          >
            <div
              className="sc-htpNat idItfP"
            >
              Postcode
            </div>
            <div
              className="sc-htpNat idItfP"
            >
              <strong>
                LS1 XXX
              </strong>
            </div>
          </div>
          <div
            className="sc-bwzfXH ijiype"
          >
            <div
              className="sc-htpNat idItfP"
            >
              Country
            </div>
            <div
              className="sc-htpNat idItfP"
            >
              <strong>
                England
              </strong>
            </div>
          </div>
          <div
            className="sc-bwzfXH ijiype"
          >
            <div
              className="sc-htpNat idItfP"
            >
              Comment
            </div>
            <div
              className="sc-htpNat idItfP"
            >
              <strong>
                comment text goes here
              </strong>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="col-md-6"
    >
      <div
        className="row"
      >
        <h3
          className="heading-medium"
        >
          Aliases
        </h3>
      </div>
      <div>
         -- 
      </div>
      <div
        className="row"
      >
        <h3
          className="heading-medium"
        >
          Identifiers
        </h3>
      </div>
      <div
        className="sc-bwzfXH ijiype"
      >
        <div
          className="sc-htpNat idItfP"
        >
          PNC number
        </div>
        <div
          className="sc-htpNat idItfP"
        >
          <strong>
            --
          </strong>
        </div>
      </div>
      <div
        className="sc-bwzfXH ijiype"
      >
        <div
          className="sc-htpNat idItfP"
        >
          CRO number
        </div>
        <div
          className="sc-htpNat idItfP"
        >
          <strong>
            --
          </strong>
        </div>
      </div>
      <div
        className="sc-bwzfXH ijiype"
      >
        <div
          className="sc-htpNat idItfP"
        >
          National insurance number
        </div>
        <div
          className="sc-htpNat idItfP"
        >
          <strong>
            --
          </strong>
        </div>
      </div>
      <div
        className="sc-bwzfXH ijiype"
      >
        <div
          className="sc-htpNat idItfP"
        >
          Driving licence number
        </div>
        <div
          className="sc-htpNat idItfP"
        >
          <strong>
            --
          </strong>
        </div>
      </div>
      <div
        className="sc-bwzfXH ijiype"
      >
        <div
          className="sc-htpNat idItfP"
        >
          Home office reference number
        </div>
        <div
          className="sc-htpNat idItfP"
        >
          <strong>
            --
          </strong>
        </div>
      </div>
      <div
        className="row"
      >
        <h3
          className="heading-medium"
        >
          Contacts
        </h3>
      </div>
      <div
        className="add-gutter-margin-bottom"
      >
        <div>
          <div
            className="row border-bottom-line"
          >
            <div
              className="col-lg-6 col-xs-6"
            >
              <span>
                Next of kin
              </span>
            </div>
            <div
              className="col-lg-6 col-xs-6"
            >
              <b>
                 No next of kin identified 
              </b>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="col-md-12"
    >
      <h3
        className="heading-medium"
      >
        Physical characteristics
      </h3>
    </div>
  </div>
  <div
    className="desktop"
  >
    <div
      className="row border-bottom-line"
    >
      <div
        className="col-md-3 col-xs-6"
      >
        <span>
          Height
        </span>
      </div>
      <div
        className="col-md-3 col-xs-6"
      >
        <strong>
          <span>
             
            1.75 metres
             
          </span>
        </strong>
      </div>
      <div
        className="col-md-3 col-xs-6"
      >
        <span>
          Weight
        </span>
      </div>
      <div
        className="col-md-3 col-xs-6"
      >
        <strong>
          <span>
             
            76 kg
             
          </span>
        </strong>
      </div>
    </div>
  </div>
  <div
    className="mobile"
  >
    <div
      className="row border-bottom-line"
    >
      <div
        className="col-md-3 col-xs-6"
      >
        <span>
          Height
        </span>
      </div>
      <div
        className="col-md-3 col-xs-6"
      >
        <strong>
          <span>
             
            1.75 metres
             
          </span>
        </strong>
      </div>
    </div>
    <div
      className="row border-bottom-line"
    >
      <div
        className="col-md-3 col-xs-6"
      >
        <span>
          Weight
        </span>
      </div>
      <div
        className="col-md-3 col-xs-6"
      >
        <strong>
          <span>
             
            76 kg
             
          </span>
        </strong>
      </div>
    </div>
  </div>
  <div
    className="desktop"
  >
    <div
      className="row border-bottom-line"
    >
      <div>
        <div
          className="col-md-3 col-xs-6"
        >
          <span>
            Hair Colour
          </span>
        </div>
        <div
          className="col-md-3 col-xs-6"
        >
          <strong>
            <span>
               
              QBNQB
               
            </span>
          </strong>
        </div>
      </div>
      <div>
        <div
          className="col-md-3 col-xs-6"
        >
          <span>
            Right Eye Colour
          </span>
        </div>
        <div
          className="col-md-3 col-xs-6"
        >
          <strong>
            <span>
               
              AIAI
               
            </span>
          </strong>
        </div>
      </div>
    </div>
    <div
      className="row border-bottom-line"
    >
      <div>
        <div
          className="col-md-3 col-xs-6"
        >
          <span>
            Build
          </span>
        </div>
        <div
          className="col-md-3 col-xs-6"
        >
          <strong>
            <span>
               
              YEkYEk
               
            </span>
          </strong>
        </div>
      </div>
    </div>
  </div>
  <div
    className="mobile"
  >
    <div>
      <div>
        <div
          className="row border-bottom-line col-md-3 col-xs-6"
        >
          <span>
            Hair Colour
          </span>
        </div>
        <div
          className="row border-bottom-line col-md-3 col-xs-6"
        >
          <strong>
            <span>
               
              QBNQB
               
            </span>
          </strong>
        </div>
      </div>
      <div>
        <div
          className="row border-bottom-line col-md-3 col-xs-6"
        >
          <span>
            Right Eye Colour
          </span>
        </div>
        <div
          className="row border-bottom-line col-md-3 col-xs-6"
        >
          <strong>
            <span>
               
              AIAI
               
            </span>
          </strong>
        </div>
      </div>
    </div>
    <div>
      <div>
        <div
          className="row border-bottom-line col-md-3 col-xs-6"
        >
          <span>
            Build
          </span>
        </div>
        <div
          className="row border-bottom-line col-md-3 col-xs-6"
        >
          <strong>
            <span>
               
              YEkYEk
               
            </span>
          </strong>
        </div>
      </div>
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="col-xs-12"
    >
      <h3
        className="heading-medium"
      >
        Distinguishing marks
      </h3>
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="col-md-6"
    >
      <div
        className="sc-bwzfXH ijiype"
      >
        <div
          className="sc-htpNat idItfP"
        >
          Type
        </div>
        <div
          className="sc-htpNat idItfP"
        >
          <strong>
            Tattoo
          </strong>
        </div>
      </div>
      <div
        className="sc-bwzfXH ijiype"
      >
        <div
          className="sc-htpNat idItfP"
        >
          Body Part
        </div>
        <div
          className="sc-htpNat idItfP"
        >
          <strong>
            Torso
          </strong>
        </div>
      </div>
      <div
        className="sc-bwzfXH ijiype"
      >
        <div
          className="sc-htpNat idItfP"
        >
          Comment
        </div>
        <div
          className="sc-htpNat idItfP"
        >
          <strong>
            INIgBKyJIPuINIgBKyJIP
          </strong>
        </div>
      </div>
    </div>
  </div>
</div>
`;
